src/components/MultiGestureCanvas/types.ts

```diff
@@ -0,0 +1,50 @@
+import type {SharedValue} from 'react-native-reanimated';
+
+/** Dimensions of the canvas rendered by the MultiGestureCanvas */
+type CanvasSize = {
+    width: number;
+    height: number;
+};
+
+/** Dimensions of the content passed to the MultiGestureCanvas */
+type ContentSize = {
+    width: number;
+    height: number;
+};
+
+/** Range of zoom that can be applied to the content by pinching or double tapping. */
+type ZoomRange = {
+    min: number;
+    max: number;
+};
+
+/** Triggered whenever the scale of the MultiGestureCanvas changes */
+type OnScaleChangedCallback = (zoomScale: number) => void;
+
+/** Triggered when the canvas is tapped (single tap) */
+type OnTapCallback = (() => void) | undefined;
+
+/** Types used of variables used within the MultiGestureCanvas component and it's hooks */
+type MultiGestureCanvasVariables = {
+    canvasSize: CanvasSize;
+    contentSize: ContentSize;
+    zoomRange: ZoomRange;
+    minContentScale: number;
+    maxContentScale: number;
+    isPagerSwiping: SharedValue<boolean>;
+    zoomScale: SharedValue<number>;
+    totalScale: SharedValue<number>;
+    pinchScale: SharedValue<number>;
+    offsetX: SharedValue<number>;
+    offsetY: SharedValue<number>;
+    panTranslateX: SharedValue<number>;
+    panTranslateY: SharedValue<number>;
+    pinchTranslateX: SharedValue<number>;
+    pinchTranslateY: SharedValue<number>;
+    stopAnimation: () => void;
+    reset: (animated: boolean, callback: () => void) => void;
+    onTap: OnTapCallback | undefined;
+    onScaleChanged?: OnScaleChangedCallback;
```

Comment:
I think this contradicts with https://github.com/Expensify/App/pull/33756#discussion_r1455116235.

Reply:
Since `MultiGestureCanvasVariables` is only used by hooks, can we make all the fields optional? And in each hook we use the types that we need and make them required i.e. `Required<Pick<MultiGestureCanvasVariables, ...>>`

Reply:
Since we just use `Pick` in the hooks to get the relevant type of the props from `MultiGestureCanvasVariables`, i'd rather leave it as optional. This way in e.g. `usePinchGesture`, we can just do this:

```ts
type UsePinchGestureProps = Pick<
    MultiGestureCanvasVariables,
    'canvasSize' | 'zoomScale' | 'zoomRange' | 'offsetX' | 'offsetY' | 'pinchTranslateX' | 'pinchTranslateY' | 'pinchScale' | 'isPagerSwiping' | 'stopAnimation' | 'onScaleChanged'
>;
```

Reply:
This way, `onScaleChanged` wouldn't be able to be `undefined`, which it is though.